lnwallet/channel.go

```diff
@@ -1882,22 +1924,12 @@ func (lc *LightningChannel) ChannelPoint() *wire.OutPoint {
 	return lc.channelState.ChanID
 }
 
-// addHTLC adds a new HTLC to the passed commitment transaction. One of four
-// full scripts will be generated for the HTLC output depending on if the HTLC
-// is incoming and if it's being applied to our commitment transaction or that
-// of the remote node's. Additionally, in order to be able to efficiently
-// locate the added HTLC on the commitment transaction from the
-// PaymentDescriptor that generated it, the generated script is stored within
-// the descriptor itself.
-func (lc *LightningChannel) addHTLC(commitTx *wire.MsgTx, ourCommit bool,
-	paymentDesc *PaymentDescriptor, revocation [32]byte, delay uint32,
-	isIncoming bool) error {
-
-	localKey := lc.channelState.OurCommitKey
-	remoteKey := lc.channelState.TheirCommitKey
-	timeout := paymentDesc.Timeout
-	rHash := paymentDesc.RHash
-
+// genHTLCP2WSHPkScript generates the proper P2WSH public key scripts for the
+// HTLC output modified by two-bits denoting if this is an incoming HTLC, and
+// if the HTLC is being applied to their commitment transaction or ours.
+func (lc *LightningChannel) genHTLCP2WSHPkScript(isIncoming, ourCommit bool,
```

Comment:
This method name is a bit of a mouthful, naming recommendation: `genHltcScript`.